src/lib/backend-api.ts

```diff
@@ -28,20 +28,23 @@ import {
   CreateAuthTokenBackendResponse,
   DeleteAuthTokenBackendResponse
 } from 'response'
-import request, { buildQueryParamString } from './utils/request'
-import { Currency } from '../types/domain'
-import * as bitcoinBackendFactory from './bitcoin/bitcoin-backend-api';
+import { buildQueryParamString, createHttpClient, HttpClient } from './utils/httpClient'
+import { Currency } from '..'
+import * as bitcoinBackendFactory from './bitcoin/bitcoin-backend-api'
 
 export interface SakiewkaBackend {
   core: CoreBackendApi,
   [Currency.BTC]: bitcoinBackendFactory.BitcoinBackendApi,
   [Currency.BTG]: bitcoinBackendFactory.BitcoinBackendApi
 }
 
-export const backendFactory = (backendApiUrl: string): SakiewkaBackend => {
-  const backendApi = create(backendApiUrl)
-  const btcBackendApi = bitcoinBackendFactory.withCurrency(backendApiUrl, Currency.BTC)
-  const btgBackendApi = bitcoinBackendFactory.withCurrency(backendApiUrl, Currency.BTG)
+export type CorrelationIdGetter = () => string
```

Comment:
a czemu tak właściwie to jest `() => string` a nie po prostu `string`? Bo tworzymy dla każdego requesto to na nowo, nie?

Reply:
Tak, dokładnie

Reply:
Nooo ... jak tworzymy na nowo, to czemu przekazujemy () => string a nie konkretne correlation id?